### ISSUE TITLE
Expression of type "type[Literal['foo']]" is incompatible with declared type "type[Literal['foo']]"

### ISSUE THREAD

**Issue by arjenzorgdoc** [microsoft/pyright#8067]
### Minimal repo:
 
```python
from typing import Literal

x: type[Literal["foo"]] = Literal["foo"]
```

```
$ PYRIGHT_PYTHON_FORCE_VERSION=latest pyright test.py 
/tmp/test.py
  /tmp/test.py:3:27 - error: Expression of type "type[Literal['foo']]" is incompatible with declared type "type[Literal['foo']]"
    Type "type[Literal]" is incompatible with type "type[Literal['foo']]" (reportAssignmentType)
1 error, 0 warnings, 0 informations 
```

### Code example:

```python
def get[T: LiteralString](t: type[T], default: T) -> T:
    ... # return something from T or:
    return default


get(Literal["foo", "bar"], "bar")
```

That used to work in 1.1.361:

```
$ PYRIGHT_PYTHON_FORCE_VERSION=1.1.361 pyright test.py 
WARNING: there is a new pyright version available (v1.1.361 -> v1.1.365).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

0 errors, 0 warnings, 0 informations 
```

But fails in 1.1.362:

```
$ PYRIGHT_PYTHON_FORCE_VERSION=1.1.362 pyright test.py 
WARNING: there is a new pyright version available (v1.1.362 -> v1.1.365).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

/tmp/test.py
  /tmp/test.py:3:34 - error: Expression of type "type[Literal['foo', 'bar']]" is incompatible with declared type "type[Literal['foo', 'bar']]"
    Type "UnionType" is incompatible with type "type[Literal['foo', 'bar']]"
      Type "UnionType" is incompatible with type "type[Literal['foo']]"
      Type "UnionType" is incompatible with type "type[Literal['bar']]" (reportAssignmentType)
  /tmp/test.py:11:5 - error: Argument of type "type[Literal['foo', 'bar']]" cannot be assigned to parameter "t" of type "type[T@get]" in function "get"
    Type "UnionType" is incompatible with type "type[Literal['bar']]" (reportArgumentType)
2 errors, 0 warnings, 0 informations 
```

**Comment by erictraut** [comment]
`Literal` is a special form intended to be used in a type expression (i.e. an annotation), but you're using it here within a value expression. The "type" of a special form when used in a value expression is currently undefined by the typing spec.

There is a [draft PEP](https://discuss.python.org/t/typeform-spelling-for-a-type-annotation-object-at-runtime) under development that will clarify this behavior.

For now, I don't plan to make any changes to accommodate this usage. Once that PEP is a bit further along, I'll implement the functionality in conformance with the spec. I'm going to close this issue for now.
